extensions/amp-bind/0.1/test/test-bind-expr.js

```diff
@@ -0,0 +1,430 @@
+    expect(evaluateBindExpr('3 * 4')).to.equal(12);
+    expect(evaluateBindExpr('4 / 5')).to.equal(0.8);
+    expect(evaluateBindExpr('5 % 4')).to.equal(1);
+    expect(evaluateBindExpr('1 / 0')).to.equal(Infinity);
+  });
+
+  it('should evaluate comparison operations', () => {
+    expect(evaluateBindExpr('2 > 1')).to.be.true;
+    expect(evaluateBindExpr('1 > 1')).to.be.false;
+    expect(evaluateBindExpr('1 >= 1')).to.be.true;
+    expect(evaluateBindExpr('1 >= 2')).to.be.false;
+    expect(evaluateBindExpr('1 < 2')).to.be.true;
+    expect(evaluateBindExpr('0 < 0')).to.be.false;
+    expect(evaluateBindExpr('1 <= 1')).to.be.true;
+    expect(evaluateBindExpr('1 <= 0')).to.be.false;
+    expect(evaluateBindExpr('0 == 1')).to.be.false;
+    expect(evaluateBindExpr('1 == 1')).to.be.true;
+    expect(evaluateBindExpr('0 != 1')).to.be.true;
+    expect(evaluateBindExpr('1 != 1')).to.be.false;
+  });
+
+  it('should evaluate logical operations', () => {
+    expect(evaluateBindExpr('!false')).to.be.true;
+    expect(evaluateBindExpr('true && true')).to.be.true;
+    expect(evaluateBindExpr('true && false')).to.be.false;
+    expect(evaluateBindExpr('false && false')).to.be.false;
+    expect(evaluateBindExpr('true || true')).to.be.true;
+    expect(evaluateBindExpr('true || false')).to.be.true;
+    expect(evaluateBindExpr('false || false')).to.be.false;
+  });
+
+  it('should respect arithmetic operator precedence', () => {
+    expect(evaluateBindExpr('-1 + 2')).to.equal(1);
+    expect(evaluateBindExpr('1 - -0.5')).to.equal(1.5);
+    expect(evaluateBindExpr('1 + -2 * 3')).to.equal(-5);
+    expect(evaluateBindExpr('1 / 2 - 3')).to.equal(-2.5);
+    expect(evaluateBindExpr('4 % 3 - 2 * 1')).to.equal(-1);
+  });
+
+  it('should respect comparison operator precedence', () => {
+    expect(evaluateBindExpr('true == 2 > 1')).to.equal(true);
+    expect(evaluateBindExpr('true == 2 >= 1')).to.equal(true);
+    expect(evaluateBindExpr('true == 2 < 1')).to.equal(false);
+    expect(evaluateBindExpr('true == 2 <= 1')).to.equal(false);
+    expect(evaluateBindExpr('1 > 2 == true')).to.equal(false);
+    expect(evaluateBindExpr('1 >= 2 == true')).to.equal(false);
+    expect(evaluateBindExpr('1 < 2 == true')).to.equal(true);
+    expect(evaluateBindExpr('1 <= 2 == true')).to.equal(true);
+  });
+
+  it('should respect logical operator precedence', () => {
+    expect(evaluateBindExpr('!false && true')).to.be.true;
+    expect(evaluateBindExpr('false || !true')).to.be.false;
+    expect(evaluateBindExpr('true && false || true')).to.be.true;
+    expect(evaluateBindExpr('true && false == false')).to.be.true;
+    expect(evaluateBindExpr('false || false == true')).to.be.false;
+    expect(evaluateBindExpr('false == !true')).to.be.true;
+  });
+
+  it('should support strings', () => {
+    expect(evaluateBindExpr('"a"')).to.equal('a');
+    expect(evaluateBindExpr('"a".length')).to.equal(1);
+    expect(evaluateBindExpr('"a" + "b"')).to.equal('ab');
```

Comment:
I'm ok deferring. But we need to see the most prevalent cases while we are implementing it toward MVP. If it's URL construction, I believe, we need to do it for MVP.